retrodep/gosource.go

```diff
@@ -533,19 +533,23 @@ func (src GoSource) RepoPathForImportPath(importPath string) (*RepoPath, error)
 
 	// No replacement found, use the import pth as-is
 	r, err := vcs.RepoRootForImportPath(importPath, false)
-	if err != nil && strings.ContainsRune(importPath, '_') {
-		// gopkg.in gives bad responses for paths like
-		// gopkg.in/foo/bar.v2/_examples/chat1
-		// because of the underscore. Remove it and try again.
-		u := strings.Index(importPath, "_")
-		importPath = path.Dir(importPath[:u])
-		r, nerr := vcs.RepoRootForImportPath(importPath, false)
-		if nerr == nil {
-			return &RepoPath{RepoRoot: *r}, nil
-		}
+	if err == nil {
+		return &RepoPath{RepoRoot: *r}, err
 	}
 
-	return &RepoPath{RepoRoot: *r}, err
+	if !strings.ContainsRune(importPath, '_') {
```

Comment:
I think we'd want a comment after that early return noting that this is the now the error path. Usually an early return would be for an error, with the continuation being the 'happy path', so without a comment this might be harder to read at first glance.

Reply:
Chaning this to 
if <no error> {}
else {}

This should be more readable.